amplify/backend/function/dogfunction/src/updateDog.js

```diff
@@ -0,0 +1,48 @@
+/* Amplify Params - DO NOT EDIT
+You can access the following resource attributes as environment variables from your Lambda function
+var environment = process.env.ENV
+var region = process.env.REGION
+var storageDogtableName = process.env.STORAGE_DOGTABLE_NAME
+var storageDogtableArn = process.env.STORAGE_DOGTABLE_ARN
+
+Amplify Params - DO NOT EDIT */
+
+var AWS = require('aws-sdk');
+var region = process.env.REGION
+AWS.config.update({region: region});
+var documentClient = new AWS.DynamoDB.DocumentClient({region: region});
+var ddb_table_name = process.env.STORAGE_DOGTABLE_NAME
+
+
+function updateDog(event, callback) { 
+
+    const input = event.arguments.input
+
+    var params = {
+        TableName: ddb_table_name,
+        Key: {id: input.id},
+        UpdateExpression: "set firstName = :fn, lastName = :ln, breed = :br, dob = :dob, sex = :sex, bio = :bio",
```

Comment:
@ZhuochaoZhong can you explain this to me?

Reply:
I created the update functions for user and dog, but in fact, we only used the create functions while updating information in the front end. When you create a user/dog with the same id in the table, it would overwrite the old user/dog entry. So we just use the create functions for creating and updating user/dog information for the front-end.

However, if you think we need to separate these two functions for the best practice, we could use this update function by passing in the id and the new attributes, it would update the table as well.